File: Harvard Assignments/readability.py
from cs50 import get_string

def main():
        while True:
            text = get_string("text:  ")
            break

        letters = 0
        words = 1
        sentences = 0

        for i in range(len(text)):
            if text[i].isalpha():
                        letters +=1
            if text[i].isspace():
                            words +=1
            if  text[i] == '.' or text[i] == '!' or text[i] == '?':
                            sentences +=1

        L =float( letters * 100 / words )
        S = float (sentences * 100 / words)
        index = 0.0588 * L - 0.296 * S - 15.8
        K= round(index)

        if K>= 1 and K <= 16:
           print("Grade:", K )
        elif  K < 1:
           print("Before Grade 1")
        elif K > 16:
           print("Grade 16+")

main()

File: Harvard Assignments/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <ctype.h>
#include <string.h>

int letters = 0;
int words = 1;
int sentences = 0;

int main(void)


{
    string text = get_string("Text: ");


    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters ++;
        }
        else  if (isspace(text[i]))
        {
            words ++;
        }
        else  if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences ++;
        }
    }

    float L  = (float)letters  * 100 / words ;
    float S = (float) sentences   * 100 / words;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 1 && index <= 16)
    {
        printf("Grade %i\n", index);
    }
    else if (index < 1)

    {
        printf("Before Grade 1\n");
    }

    else  if (index > 16)
    {
        printf("Grade 16+\n");
    }



}


